design/one-pager-managed-resource-api-design.md

```diff
@@ -246,6 +263,20 @@ type EmbeddedStructStatus struct {
 }
 ```
 
+#### Sensitive Input Fields
+
+Some cloud services can take sensitive input such as passwords, certificates or
+tokens. However, exposing those fields on the CR is not the best way to handle it.
+For such fields, the field name should be `<field-name>SecretRef` and the type
+of that field should be `SecretKeySelector`([from crossplane-runtime]). The controller
+should fetch the value from the given secret and populate the field in the provider's
+corresponding SDK object.
+
+In case the secret reference is not provided **and** the field is a required one,
+such as password, the controller should generate it randomly. Then it has to make
+sure the generated value ends up in the connection details secret that is published
+by the controller so that no information is lost or not exposed to the user.
```

Comment:
Should we include a requirement in this document that such fields must document this behaviour in their GoDoc comment?